packages/ecc-utils-design/src/components/wrapper/wrapper.ts

```diff
@@ -1,18 +1,56 @@
-import { attr } from "@microsoft/fast-element";
-import { FoundationElement } from "@microsoft/fast-foundation";
-import { registerDesignTokens, setToken } from "../../design-system/index.js";
-import { backgroundColor, textColor } from "../../design-system/tokens.js";
+import { attr, observable } from '@microsoft/fast-element';
+import { FoundationElement, CSSDesignToken } from '@microsoft/fast-foundation';
+import {
+  createToken,
+  registerDesignTokens,
+  setToken,
+} from '../../design-system/index.js';
+import allTokens from '../../design-system/tokens.js';
+
+const isObject = (o: unknown) =>
+  o instanceof Object && o.constructor === Object;
+
+const camelize = (s: string) => s.replace(/-./g, (x) => x[1].toUpperCase());
 
 export class Wrapper extends FoundationElement {
   // config: handle passing the design system config
-  @attr config = "";
+  @attr config: any = '';
+  @observable locallyStoredTokens: Record<string, CSSDesignToken<string>> = {};
+
+  private storeTokensLocallyForTesting(
+    registeredTokenName: string,
+    registeredTokenValue: CSSDesignToken<string>
+  ) {
+    this.locallyStoredTokens[registeredTokenName] = registeredTokenValue;
+  }
 
   connectedCallback() {
     super.connectedCallback();
     registerDesignTokens(this);
-    const configJSON = JSON.parse(this.config);
-    if (configJSON.backgroundColor)
-      setToken(this, configJSON.backgroundColor, backgroundColor);
-    if (configJSON.textColor) setToken(this, configJSON.textColor, textColor);
+
+    if (!this.config) return;
+    let configJSON = this.config;
+
+    if (!isObject(configJSON)) configJSON = JSON.parse(this.config);
+
+    Object.keys(configJSON).forEach((i) => {
+      // add case to not register when object name is "default"
+      if (isObject(configJSON[i])) {
+        Object.keys(configJSON[i]).forEach((j) => {
+          const newToken = createToken(`${i}-${j}`);
```

Comment:
At present, this generates tokens for every `name` and `property`, such as `primary-color`. It could be beneficial to establish a method for using a single token, such as `background-color`, and assigning different values to it for different wrapper roots.
I think that will be a much cleaner way of doing it & will not bloat our components in case of many different `name` props.

Reply:
yeah, that makes sense. I'm actually tinkering with a different implementation that's much more efficient, there just isn't much documentation so i've been sort of reaching around in the dark. but i think i have something.